web/src/actions/apd.js

```diff
@@ -38,3 +48,100 @@ export const fetchApdDataIfNeeded = () => (dispatch, getState) => {
 
   return null;
 };
+
+export const saveApd = () => (dispatch, state) => {
+  dispatch(requestSave());
+
+  const {
+    apd: { data: updatedApd },
+    activities: { byId: activitiesByID }
+  } = state();
+
+  const apd = {
+    activities: [],
+    keyPersonnel: updatedApd.keyPersonnel.map(p => ({
+      title: p.title,
+      description: p.desc,
+      years: Object.keys(p.years).map(year => ({
+        year,
+        cost: p.years[year].amt,
+        fte: p.years[year].perc
+      }))
+    })),
+    narrativeHIE: updatedApd.hieNarrative,
+    narrativeHIT: updatedApd.hitNarrative,
+    narrativeMMIS: updatedApd.mmisNarrative,
+    programOverview: updatedApd.overview,
+    years: updatedApd.years
+  };
+
+  Object.keys(activitiesByID).forEach(id => {
+    const activity = activitiesByID[id];
+    apd.activities.push({
+      name: activity.name,
+      types: activity.types,
+      summary: activity.descShort,
+      description: activity.descLong,
+      alternatives: activity.altApproach,
+      costAllocationMethodology: activity.costAllocateDesc,
+      otherFundingSources: {
+        description: activity.otherFundingDesc,
+        amount: activity.otherFundingAmt
+      },
+      goals: activity.goals.map(g => ({
+        description: g.desc,
+        objectives: [{ description: g.obj }] // TODO - objective should mape 1:1 to goals, instead of being an array (needs backend change first)
```

Comment:
See #550. Once we figure out how the API will send/receive goals and objectives, we should update this.